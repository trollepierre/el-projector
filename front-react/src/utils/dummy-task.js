const dummyTask = (task) => ({
  id: 1,
  name: 'Build el-projector',
  description: 'more data about how to do it',
  points: 999, // given by server
  createdDate: '19/05/2019',
  endDate: undefined, // must be done before this date
  optimalDate: undefined, // user would do it at this date
  isDone: false,
  doneDate: undefined, // for log record
  priority: 'HIGHLY_IMPORTANT', // Enum with 5 levels (inherited by default, but can be updated)
  benefit: 'ORGASM_HAPPINESS', // Enum with 5 levels
  neededTime: 'ONE_MONTH', // Enum
  epicId: 'A1', // Yearly happiness epic
  neededMaterial: ['MacTo', 'iMoi', 'MacNo', 'home'],
  subTaskIds: ['TASK10', 'TASK11', 'TASK12'], // checklist of subtasks that depends on it
  masterTaskId: 'TASK0', // this task depends on the master one
  ...task,
})

export { dummyTask }
