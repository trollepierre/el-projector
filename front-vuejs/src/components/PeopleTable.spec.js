import Vue from 'vue'
import TaskTable from './TaskTable.vue'
import Task from './task/Task.vue'
import CreateTask from './create-task/CreateTask.vue'
import EditTask from './edit-task/EditTask.vue'
import tasksApi from '../api/tasks'
import { dummyTask } from '../utils/test/dummy-task'
import meetingApi from '../api/meeting'

describe('TaskTable.vue', () => {
  const firstTask = dummyTask()
  const tasks = [firstTask]

  beforeEach(() => {
    // TODO remove it
    Vue.config.silent = true
    tasksApi.fetch = jest.fn()
  })

  describe('template', () => {
    it('should contain a row to create new task', () => {
      // When
      const wrapper = shallowMount(TaskTable, { data: () => ({ tasks }) })

      // Then
      expect(wrapper.find(CreateTask).props()).toEqual({ newTask: {} })
    })

    describe('when tasks contain one task', () => {
      it('should match snapshot', () => {
        // When
        const wrapper = shallowMount(TaskTable, { data: () => ({ tasks }) })

        // Then
        expect(wrapper.element).toMatchSnapshot()
      })

      it('should contain a row with a task', () => {
        // When
        const wrapper = shallowMount(TaskTable, { data: () => ({ tasks }) })

        // Then
        expect(wrapper.find(Task).props()).toEqual({ birthday: '', task: firstTask })
      })
    })

    describe('when one task is being edited', () => {
      it('should match snapshot', () => {
        // Given
        const editTask = dummyTask()

        // When
        const wrapper = shallowMount(TaskTable, { data: () => ({ tasks: [editTask], editTask }) })

        // Then
        expect(wrapper.element).toMatchSnapshot()
      })

      it('should contain a row with an editable task', () => {
        // Given
        const editTask = dummyTask()

        // When
        const wrapper = shallowMount(TaskTable, { data: () => ({ tasks: [editTask], editTask }) })

        // Then
        expect(wrapper.find(EditTask).props()).toEqual({ task: firstTask })
      })
    })
  })

  describe('event', () => {
    describe('Task', () => {
      describe('on click on seen button', () => {
        it('should add meeting to api', () => {
          // Given
          const wrapper = shallowMount(TaskTable, { data: () => ({ tasks }) })
          meetingApi.meet = jest.fn()

          // When
          wrapper.find(Task).vm.$emit('meet', firstTask)

          // Then
          expect(meetingApi.meet).toHaveBeenCalledWith(firstTask)
        })
      })
      describe('on click on later button', () => {
        it('should differ meeting to api', () => {
          // Given
          const wrapper = shallowMount(TaskTable, { data: () => ({ tasks }) })
          meetingApi.differ = jest.fn()

          // When
          wrapper.find(Task).vm.$emit('differ', firstTask)

          // Then
          expect(meetingApi.differ).toHaveBeenCalledWith(firstTask)
        })
      })
      describe('on click on nope button', () => {
        it('should reject meeting to api', () => {
          // Given
          const wrapper = shallowMount(TaskTable, { data: () => ({ tasks }) })
          meetingApi.reject = jest.fn()

          // When
          wrapper.find(Task).vm.$emit('reject', firstTask)

          // Then
          expect(meetingApi.reject).toHaveBeenCalledWith(firstTask)
        })
      })
    })
  })
})
