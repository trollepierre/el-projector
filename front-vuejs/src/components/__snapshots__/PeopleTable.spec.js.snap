// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTable.vue template when one task is being edited should match snapshot 1`] = `
<div
  class="my-table"
>
  <table>
    <tableheader-stub />

    <tbody>
      <createtask-stub
        newtask="[object Object]"
      />

      <div>
        <edittask-stub
          task="[object Object]"
        />
      </div>
    </tbody>
  </table>
</div>
`;

exports[`TaskTable.vue template when tasks contain one task should match snapshot 1`] = `
<div
  class="my-table"
>
  <table>
    <tableheader-stub />

    <tbody>
      <createtask-stub
        newtask="[object Object]"
      />

      <div>
        <task-stub
          birthday=""
          task="[object Object]"
        />
      </div>
    </tbody>
  </table>
</div>
`;
